applications/TopologyOptimizationApplication/custom_strategies/custom_schemes/residualbased_incrementalupdate_static_simp_scheme.h

```diff
@@ -0,0 +1,266 @@
+/*@{ */
+
+/*@} */
+
+
+/**@name  Enum's */
+/*@{ */
+
+
+/*@} */
+/**@name  Functions */
+/*@{ */
+
+
+/*@} */
+/**@name Kratos Classes */
+/*@{ */
+
+template<class TSparseSpace,class TDenseSpace >
+class ResidualBasedIncrementalUpdateStaticSIMPScheme : public ResidualBasedIncrementalUpdateStaticScheme<TSparseSpace,TDenseSpace>
+{
+
+public:
+    /**@name Type Definitions */
+    /*@{ */
+
+    KRATOS_CLASS_POINTER_DEFINITION( ResidualBasedIncrementalUpdateStaticSIMPScheme );
+
+    typedef ResidualBasedIncrementalUpdateStaticScheme<TSparseSpace,TDenseSpace> BaseType;
+
+    typedef typename BaseType::TDataType TDataType;
+
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+
+    typedef typename BaseType::TSystemMatrixType TSystemMatrixType;
+
+    typedef typename BaseType::TSystemVectorType TSystemVectorType;
+
+    typedef typename BaseType::LocalSystemVectorType LocalSystemVectorType;
+    typedef typename BaseType::LocalSystemMatrixType LocalSystemMatrixType;
+
+
+    /// The definition of the vector containing the equation ids
+    typedef Element::EquationIdVectorType                              EquationIdVectorType;
+
+    /*@} */
+    /**@name Life Cycle
+    */
+    /*@{ */
+
+    /** Constructor.
+    */
+    ResidualBasedIncrementalUpdateStaticSIMPScheme()
+        : ResidualBasedIncrementalUpdateStaticScheme<TSparseSpace,TDenseSpace>()
+    {}
+
+    /** Destructor.
+    */
+    ~ResidualBasedIncrementalUpdateStaticSIMPScheme() override {} 
+
+
+    /*@} */
+    /**@name Operators
+    */
+    /*@{ */
+
+    // ---------------------------------------------------------------------------------------------------------------------------------------------
+    // --------------------------------- UPDATE LHS AND RHS ----------------------------------------------------------------------------------------
+    // ---------------------------------------------------------------------------------------------------------------------------------------------
+
+    /// This function calculates a new Youngs Modulus based on the densities and multiplies it into the
+    /// LHS and RHS contributions of the complete system
+
+
+
+    void CalculateSystemContributions(
+        Element& rCurrentElement,
+        LocalSystemMatrixType& rLHSContribution,
+        LocalSystemVectorType& rRHSContribution,
+        EquationIdVectorType& rEquationId,
+        const ProcessInfo& rCurrentProcessInfo
+        ) override
+    {
+        KRATOS_TRY
+        //Initializing the non linear iteration for the current element
+        rCurrentElement.InitializeNonLinearIteration(rCurrentProcessInfo);
+
+        //basic operations for the element considered
+        rCurrentElement.CalculateLocalSystem(rLHSContribution,rRHSContribution,rCurrentProcessInfo);
+        
+
+        //Determine the new Youngs Modulus based on the assigned new density (X_PHYS)
+        double E_min     = rCurrentElement.GetValue(YOUNGS_MODULUS_MIN);
```

Comment:
```suggestion
        const double E_min     = rCurrentElement.GetValue(YOUNGS_MODULUS_MIN);
```
Idem for the following